File: app/src/main/java/com/legendsoftware/richmangoogleplaybillinglibrarytest/billing/PurchaseUpdateListener.java
package com.legendsoftware.richmangoogleplaybillinglibrarytest.billing;

import com.android.billingclient.api.BillingResult;
import com.android.billingclient.api.ProductDetails;
import com.android.billingclient.api.Purchase;
import java.util.List;

public interface PurchaseUpdateListener {

    void onProductsLoaded(List<ProductDetails> productDetailsList);

    void onPurchaseSuccess(Purchase purchase);

    void onPurchaseFailed(BillingResult billingResult, String errorMessage);

    void onPurchasePending(Purchase purchase);

    void onPurchaseCanceled();
}

File: app/src/main/java/com/legendsoftware/richmangoogleplaybillinglibrarytest/billing/PurchaseUpdateListener.kt
package com.legendsoftware.richmangoogleplaybillinglibrarytest.billing

import com.android.billingclient.api.BillingResult
import com.android.billingclient.api.Purchase

interface PurchaseUpdateListener {
    fun onProductsLoaded(productOptions: List<PurchaseOption>)
    fun onPurchaseSuccess(purchase: Purchase?)
    fun onPurchaseFailed(billingResult: BillingResult?, errorMessage: String?)
    fun onPurchasePending(purchase: Purchase?)
    fun onPurchaseCanceled()
    fun onSubscriptionStatusUpdated(purchaseToken: String?) = Unit
}
